entities/src/main/java/org/odk/collect/entities/LocalEntityUseCases.kt

```diff
@@ -104,14 +146,18 @@ object LocalEntityUseCases {
             return null
         }
 
-        return Entity.New(
-            list,
+        return ServerEntity(
             id,
             label,
             version,
-            map.toList(),
-            state = Entity.State.ONLINE,
-            trunkVersion = version
+            map
         )
     }
 }
+
+private data class ServerEntity(
```

Comment:
I'm not sure if introducing this class makes the code cleaner... Maybe I'm missing something. Why using Entity.New was bad?

Reply:
I personally found it easier to reason about parsing it into a dedicated class rather than `Entity.New` which was being "repurposed" really. For instance, a server entity (one from the CSV) doesn't have a trunk version or a branch ID and its state is always "online" conceptually. At best, it felt clunky to be setting these values, and at worst, it created risk around setting them to the wrong thing and then using that in as the object that gets saved to the DB.